src/Aspire.Hosting.Redis/RedisInsightConfigWriterHook.cs

```diff
@@ -0,0 +1,104 @@
+
+        if (!redisInstances.Any())
+        {
+            // No-op if there are no Redis resources present.
+            return;
+        }
+
+        var connectionFileDirectoryPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        if (!Directory.Exists(connectionFileDirectoryPath))
+        {
+            Directory.CreateDirectory(connectionFileDirectoryPath);
+        }
+        var connectionFilePath = Path.Combine(connectionFileDirectoryPath, "RedisInsight_connections.json");
+
+        using (var stream = new FileStream(connectionFilePath, FileMode.OpenOrCreate))
+        {
+            using var writer = new Utf8JsonWriter(stream);
+            // Need to grant read access to the config file on unix like systems.
+            if (!OperatingSystem.IsWindows())
+            {
+                File.SetUnixFileMode(connectionFilePath, UnixFileMode.UserRead | UnixFileMode.UserWrite | UnixFileMode.GroupRead | UnixFileMode.OtherRead);
+            }
+
+            writer.WriteStartArray();
+
+            foreach (var redisResource in redisInstances)
+            {
+                if (redisResource.PrimaryEndpoint.IsAllocated)
+                {
+                    var endpoint = redisResource.PrimaryEndpoint;
+                    writer.WriteStartObject();
+                    writer.WriteString("host", endpoint.ContainerHost);
+                    writer.WriteNumber("port", endpoint.Port);
+                    writer.WriteString("name", redisResource.Name);
+                    writer.WriteNumber("db", 0);
+                    writer.WriteNull("username");
+                    writer.WriteNull("password");
+                    writer.WriteString("connectionType", "STANDALONE");
+                    writer.WriteEndObject();
+                }
+
+            }
+
+            writer.WriteEndArray();
+        };
+
+        await resourceNotificationService.WaitForResourceAsync(redisInsightResource.Name, KnownResourceStates.Running, cancellationToken).ConfigureAwait(false);
+
+        var client = httpClientFactory.CreateClient();
+
+        var content = new MultipartFormDataContent();
+
+        using (var file = File.OpenRead(connectionFilePath))
+        {
+            var fileContent = new StreamContent(file);
+
+            content.Add(fileContent, "file", "RedisInsight_connections.json");
+
+            var insightEndpoint = redisInsightResource.PrimaryEndpoint;
+            var apiUrl = $"{insightEndpoint.Scheme}://{insightEndpoint.Host}:{insightEndpoint.Port}/api/databases/import";
+
+            var pipeline = new ResiliencePipelineBuilder().AddRetry(new Polly.Retry.RetryStrategyOptions
+            {
+                Delay = TimeSpan.FromSeconds(2),
+                MaxRetryAttempts = 5,
+            }).Build();
+
+            await pipeline.ExecuteAsync(async (ctx) =>
+            {
+                var response = await client.PostAsync(apiUrl, content, ctx)
```

Comment:
Is this the only way we can configure Redis Insight? I'm a bit concerned about sending HTTP requests from the AppHost. Can we instead write a file to a bind mount, like other places?

Reply:
AFIK there is no way to configure Redis Insight from file. There is some open issue about this https://github.com/RedisInsight/RedisInsight/issues/2785.
I could'nt find any better way.